docs/dev/table/streaming.md

```diff
@@ -39,17 +39,293 @@ Dynamic table
+
+- Extending the physical schema by an additional logical field
+- Replacing a physical field by a logical field (e.g. because it is not needed anymore after timestamp extraction).
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+// extract timestamp from first field and assign watermarks based on knownledge about stream
+DataStream<Tuple3<Long, String, String>> stream = inputStream.assignTimestampsAndWatermarks(...);
+
+// Option 1:
+
+// the first field has still some value and should be kept
+// declare an additional logical field as event time attribute
+Table table = tEnv.fromDataStream(stream, "UserActionTimestamp, Username, Data, UserActionTime.rowtime");
+
+
+// Option 2:
+
+// the first field has been used for timestamp extraction and is not necessary anymore
+// replace first field as logical event time attribute
+Table table = tEnv.fromDataStream(stream, "UserActionTime.rowtime, Username, Data");
+
+WindowedTable windowedTable = table.window(Tumble.over("10.minutes").on("UserActionTime").as("userActionWindow"));
+{% endhighlight %}
+</div>
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+// extract timestamp from first field and assign watermarks based on knownledge about stream
+val stream: DataStream[(Long, String, String)] = inputStream.assignTimestampsAndWatermarks(...)
+
+// Option 1:
+
+// the first field has still some value and should be kept
+// declare an additional logical field as event time attribute
+val table = tEnv.fromDataStream(stream, 'UserActionTimestamp, 'Username, 'Data, 'UserActionTime.rowtime)
+
+
+// Option 2:
+
+// the first field has been used for timestamp extraction and is not necessary anymore
+// replace first field as logical event time attribute
+val table = tEnv.fromDataStream(stream, 'UserActionTime.rowtime, 'Username, 'Data)
+
+val windowedTable = table.window(Tumble over 10.minutes on 'UserActionTime as 'userActionWindow)
+{% endhighlight %}
+</div>
+</div>
+
+#### Using a TableSource
+
+The event time attribute is defined by a `TableSource` that implements the `DefinedRowtimeAttribute` interface. The logical time attribute is appended to the physical schema defined by the return type of the `TableSource`.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+// define table source with rowtime attribute
+public class UserActionSource implements StreamTableSource<Row>, DefinedRowtimeAttribute {
+
+	@Override
+	public TypeInformation<Row> getReturnType() {
+		String[] names = new String[] {"Username" , "Data"};
+		TypeInformation[] types = new TypeInformation[] {Types.STRING(), Types.STRING()};
+		return Types.ROW(names, types);
+	}
+
+	@Override
+	public DataStream<Row> getDataStream(StreamExecutionEnvironment execEnv) {
+		// create stream 
+		// ...
+		// extract timestamp and assign watermarks based on knownledge about stream
```

Comment:
extract timestamp and assign watermarks based on knowledge of the stream
  
            
  Write